vulnerabilities/helpers.py

```diff
@@ -100,9 +118,41 @@ def requests_with_5xx_retry(max_retries=5, backoff_factor=0.5):
     return session
 
 
-def contains_alpha(string):
-    """
-    Return True if the input 'string' contains any alphabet
-    """
+def nearest_patched_package(vulnerable_packages, resolved_packages):
+    # This class is used to  get around bisect module's lack of supplying custom
+    # compartor. Get rid of this once we use python 3.10 which supports this.
+    # See https://github.com/python/cpython/pull/20556
+    class PackageURLWithVersionComparator:
+        def __init__(self, package):
+            self.package = package
+            self.version_object = version_class_by_package_type[package.type](package.version)
 
-    return any([c.isalpha() for c in string])
+        def __eq__(self, other):
+            return self.version_object == other.version_object
+
+        def __lt__(self, other):
+            return self.version_object < other.version_object
+
+    vulnerable_packages = sorted(
+        [PackageURLWithVersionComparator(package) for package in vulnerable_packages]
+    )
+    resolved_packages = sorted(
+        [PackageURLWithVersionComparator(package) for package in resolved_packages]
+    )
+
+    resolved_package_count = len(resolved_packages)
+    affected_package_with_patched_package_objects = []
```

Comment:
Why the _objects suffix?
```suggestion
    affected_package_with_patched_packages = []
```

Reply:
I'm aware of the principle of not specifying the type. The problem here is this list contains instances of `AffectedPackageWithPatchedPackage` . I don't know how to pluralise that, 

`affected_package_with_patched_packages` -> is it one one affected package with multiple patched packages
`affected_packages_with_patched_package`  -> vice versa 

This is a mess :(

Reply:
Based on other review, `affected_package` would be a better name for now.